### PR TITLE
Add bpf.cs anal plugin

### PR DESCRIPTION
**Pull request by terorie**
- [x] Mark this if you consider it ready to merge
- [x] I've added tests (optional) -- Only minimally
- [ ] I wrote some lines in the [book](https://github.com/radareorg/radare2book) (optional)

**Description**

Adds Capstone disassembler for BPF

### CODE REVIEW

**Comment by trufae** [comment]
Yay! I was about to do this in some hours! That was fast :) some comments before merging:

* asm plugins are in process of deprecation, can you move the code to be just an anal plugin? (Anal.op callback takes a mask that specifies if its suposed to disasm, esil, analyze...)
* Add the plugin to be compiled for meson builds too


Apart from that it lgtm!

File: libr/asm/p/asm_bpf_cs.c [hunk 1]
```diff
@@ -0,0 +1,62 @@
...
+
+RAsmPlugin r_asm_plugin_bpf_cs = {
+	.name = "bpf",
+	.desc = "Capstone " CAPSTONE_VERSION_STRING " BPF disassembler",
+	.license = "BSD",
+	.author = "terorie",
+	.arch = "bpf",
+	.cpus = "bpf",
+	.bits = 64,
+	.endian = R_SYS_ENDIAN_LITTLE | R_SYS_ENDIAN_BIG,
```
Comment (terorie) [inline libr/asm/p/asm_bpf_cs.c:51]: Should actually pass through endianness to Capstone

Comment (trufae) [reply libr/asm/p/asm_bpf_cs.c:51]: Yes. But i think bpf is always big endian. Isnt?

**Comment by terorie** [comment]
Got it, will fix 🙏🏻

File: libr/asm/p/asm_bpf_cs.c [hunk 2]
```diff
@@ -0,0 +1,62 @@
...
+}
+
+RAsmPlugin r_asm_plugin_bpf_cs = {
+	.name = "bpf",
+	.desc = "Capstone " CAPSTONE_VERSION_STRING " BPF disassembler",
+	.license = "BSD",
+	.author = "terorie",
+	.arch = "bpf",
+	.cpus = "bpf",
+	.bits = 64,
```
[the remark this answers is not in the document]
Comment (trufae) [reply libr/asm/p/asm_bpf_cs.c:50]: Thats up to capstone. But sure if it does 🤔

File: libr/asm/p/asm_bpf_cs.c (hunk 1 above)
Comment (terorie) [reply libr/asm/p/asm_bpf_cs.c:51]: Capstone and LLVM support both. I'm not sure what the Linux kernel does, but Solana uses little-endian.

File: libr/asm/p/asm_bpf_cs.c (hunk 2 above)
[the remark this answers is not in the document]
Comment (terorie) [reply libr/asm/p/asm_bpf_cs.c:50]: It's not really just register size – Classic BPF (32-bit GPRs) and extended BPF (64-bit GPRs) are very different architectures. https://www.kernel.org/doc/html/latest/bpf/classic_vs_extended.html
So I'm not sure if bit size is an appropriate way to switch here.  🤔

File: libr/anal/p/anal_bpf_cs.c [hunk 3]
```diff
@@ -0,0 +1,84 @@
...
+			cs_close (&handle);
+			handle = 0; // unnecessary
+		}
+		omode = mode;
+		obits = a->config->bits;
+	}
+	op->size = 8;
+	op->addr = addr;
+	if (handle == 0) {
+		ret = cs_open(CS_ARCH_BPF, mode, &handle);
```
Comment (trufae) [inline libr/anal/p/anal_bpf_cs.c:30]:
```suggestion
		ret = cs_open (CS_ARCH_BPF, mode, &handle);
```

File: libr/anal/p/anal_bpf_cs.c [hunk 4]
```diff
@@ -0,0 +1,84 @@
...
+		}
+		op->size = insn->size;
+		op->id = insn->id;
+		cs_free (insn, n);
+	}
+	return op->size;
+}
+
+RAnalPlugin r_anal_plugin_bpf_cs = {
+	.name = "bpf",
```
Comment (trufae) [inline libr/anal/p/anal_bpf_cs.c:58]:
```suggestion
	.name = "bpf.cs",
```

Comment (trufae) [reply libr/anal/p/anal_bpf_cs.c:58]: Not sure which should be the 'default' bpf disassembler. as i plan to maintain both bpf plugins, the other plugin supports assembling and esil. so let's make the cs plugin non-default for now

**Review by trufae — CHANGES_REQUESTED** [review]

File: libr/anal/p/anal_bpf_cs.c [hunk 5]
```diff
@@ -0,0 +1,84 @@
...
+#if CS_API_MAJOR >= 5
+
+static R_TH_LOCAL csh cd = 0;
+
+static int analop(RAnal *a, RAnalOp *op, ut64 addr, const ut8 *buf, int len, RAnalOpMask mask) {
+	static R_TH_LOCAL csh handle = 0;
+	static R_TH_LOCAL int omode = -1;
+	static R_TH_LOCAL int obits = 32;
+	cs_insn *insn = NULL;
+	int mode = (a->config->bits==32)? CS_MODE_BPF_CLASSIC: CS_MODE_BPF_EXTENDED;
```
Comment (trufae) [inline libr/anal/p/anal_bpf_cs.c:16]:
```suggestion
	int mode = (a->config->bits == 32)? CS_MODE_BPF_CLASSIC: CS_MODE_BPF_EXTENDED;
```

**Review by trufae — CHANGES_REQUESTED** [review]

**Pull request merged**